frontend/src/components/widgets/BaseWidget/index.tsx

```diff
@@ -15,9 +15,33 @@
  * limitations under the License.
  */
 
+import React from "react"
+import { StyledWidgetLabel } from "./styled-components"
+
 export {
   StyledWidgetInstructions,
   StyledWidgetLabel,
   StyledWidgetLabelHelp,
   StyledWidgetLabelHelpInline,
 } from "./styled-components"
+
+interface Props {
+  // Label body text. If nullsy, WidgetLabel won't show. But if empty string it will.
+  label?: string | null
+
+  // Used to specify other elements that should go inside the label container, like a help icon.
+  children?: React.ReactNode
+}
+
+export function WidgetLabel({ label, children }: Props): React.ReactElement {
+  if (label == null) {
```

Comment:
Making sure this is clear, but `label` is set to `""` in all labels, not null. Is this intentional? It has the side-effect of creating a label for every widget to maintain the vertical alignment, but...it seems weird to even check, so I am a bit confused here.

Reply:
This is useful for widgets that we insert ourselves, as opposed to the ones that come via protos.

With this change we can choose to hide labels in places where they shouldn't occupy space, like the Settings dialog box. See screenshots below.

## Before
<img width="526" alt="Screen Shot 2021-08-24 at 11 26 47 PM" src="https://user-images.githubusercontent.com/690814/130737939-651f18e5-c8c1-4eb9-8b58-434ab730c6f7.png">

## After
<img width="526" alt="Screen Shot 2021-08-24 at 11 25 16 PM" src="https://user-images.githubusercontent.com/690814/130737967-d082dd86-1455-40c8-8ed4-95333fd4e0da.png">